students/tihsle/lesson04/mailroom2.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+import sys
+
+donors =    {"William Gates, III": [65000.00,10.00],
+            "Mark Zuckerberg": [1442.25, 4000.00, 5000.00],
+            "Jeff Bezos": [877.33],
+            "Paul Allen": [400.00, 200.0000, 10.00],
+            "Joe Smithe": [1000.00, 2000.00]
+            }
+
+
+def send_thank_you():
+    #send a thank you
+
+    #which donor
+    donor = select_donor()
+
+    #how much is the gift
+    donation = input("How much is the gift?\n")
+    donors[donor].append(float(donation))
+
+    #print thank you email
+    print("Dear {},\n"
+          "\nThank you for donating!  Your generous donation will be used to fulfill our mission.\n"
+          "We look forward to updating you on how your funds are used throughout the year.\n"
+          "\nBest Regards,\n"
+          "\nCharity1\n".format(donor))
+
+
+def select_donor():
+    #select the donor for the thank you email
+
+    while True:
+        answer = input("Which donor to send Thank You to?\n"
+                         "(Type 'list' to show list of donors)\n").title()
+        if answer.lower() == "list":
+            for d in donors:
+                print("{}".format(d))
+        else:
+            exists = False
+            for donor in donors:
+                if donor == answer:
+                    exists = True
```

Comment:
this is incorrect - you should not be looping over dictionary but instead utilize dictionary look ups